src/python/zquantum/optimizers/scipy_optimizer.py

```diff
@@ -17,24 +17,34 @@ def __init__(
         self,
         method: str,
         constraints: Optional[Tuple[Dict[str, Callable]]] = None,
+        bounds: Union[
+            scipy.optimize.Bounds,
+            Sequence[Tuple[float, float]],
+            None,
+        ] = None,
```

Comment:
These other parameters use `Optional`, what do you think about using that here too? It adds a bit of extra text
```suggestion
        bounds: Optional[
            Union[
                scipy.optimize.Bounds,
                Sequence[Tuple[float, float]],
            ]
        ] = None,

Reply:
That was the first thing I tried.
But it turns out that you can't actually use `Optional` on `Union`, as `Optional` is just a shorthand for `Union[X, None]` and it throws an error. See [SO thread](https://stackoverflow.com/questions/51710037/how-should-i-use-the-optional-type-hint).

Reply:
TIL that doesn't work.